modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/HeapLockManager.java

```diff
@@ -668,8 +684,13 @@ private void track(UUID txId) {
          * @param acquirerTx Transaction which tries to acquire the lock.
          * @param holderTx Transaction which holds the lock.
          */
-        private void conflictFound(UUID acquirerTx, UUID holderTx) {
-            fireEvent(LockEvent.LOCK_CONFLICT, new LockEventParameters(acquirerTx, holderTx));
+        private boolean conflictFound(UUID acquirerTx, UUID holderTx) {
+            CompletableFuture<Void> eventResult = fireEvent(LockEvent.LOCK_CONFLICT, new LockEventParameters(acquirerTx, holderTx));
+            // No async handling is expected.
+            // TODO: https://issues.apache.org/jira/browse/IGNITE-21153
+            assert eventResult.isDone();
```

Comment:
Add a comment to the assertion.
Assert without a comment is forbidden in our code style.